addons/crm_livechat/static/tests/message_tests.js

```diff
@@ -0,0 +1,79 @@
+/* @odoo-module */
+
+import { startServer } from "@bus/../tests/helpers/mock_python_environment";
+import { start } from "@mail/../tests/helpers/test_utils";
+import { Message } from "@mail/core/common/message";
+import { makeDeferred, patchWithCleanup } from "@web/../tests/helpers/utils";
+import { click, insertText } from "@web/../tests/utils";
+
+QUnit.module("message");
+
+QUnit.test("Can open lead from internal link", async function (assert) {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({ name: "Mitchell Admin" });
+    const leadId = pyEnv["crm.lead"].create({ name: "My lead" });
+    const liveChannelId = pyEnv["im_livechat.channel"].create({ name: "YourWebsite.com" });
+    const channelId = pyEnv["discuss.channel"].create({
+        name: "Visitor, Mitchell Admin",
+        livechat_operator_id: partnerId,
+        livechat_channel_id: liveChannelId,
+        channel_type: "livechat",
+        anonymous_name: "Visitor"
+    });
+
+    const onMessageReceived = makeDeferred();
+    const { openDiscuss, env } = await start({
+        async mockRPC(route, args) {
+            if (args.method === "execute_command_lead") {
+                assert.deepEqual(args.args[0], [channelId]);
+                pyEnv["bus.bus"]._sendone(
+                    partnerId,
+                    "discuss.channel/transient_message", {
+                        body: `
+                            <span class="o_mail_notification">
+                                Create a new lead: <a href="#" data-oe-model="crm.lead" data-oe-id="${leadId}">My Lead</a>
+                            </span>`,
+                        res_id: channelId,
+                        model: "discuss.channel",
+                    }
+                );
+                onMessageReceived.resolve();
+                return true;
+            }
+        },
+    });
+
+    const onMessageLinkClicked = makeDeferred();
+    patchWithCleanup(Message.prototype, {
```

Comment:
You should use these `import { click, contains } from "@web/../tests/utils";` and find appropriate selectors at each step rather than using patches everywhere.